### PR TITLE
Enhancement/317 add date filters for adsense

### PR DESCRIPTION
**Pull request by ryanwelcher**
## Summary

Addresses issue #317

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/modules/adsense/components/dashboard/AdSenseDashboardWidget.js [hunk 1]
```diff
@@ -72,169 +72,117 @@ class AdSenseDashboardWidget extends Component {
...
+						<div className={ classnames(
+							'mdc-layout-grid__cell',
+							'mdc-layout-grid__cell--span-12',
+							wrapperClass
+						) }>
+							<Layout
+								header
+								/* translators: %s: date range */
+								title={ sprintf( __( 'Top content over the last %s', 'google-site-kit' ), currentDateRange ) }
+								headerCtaLabel={ __( 'Advanced Settings', 'google-site-kit' ) }
```
Comment (felixarntz) [inline assets/js/modules/adsense/components/dashboard/AdSenseDashboardWidget.js:153]: This is also different from the ACs.

File: assets/js/modules/adsense/components/dashboard/AdSensePerformanceWidget.js [hunk 2]
```diff
@@ -16,156 +16,134 @@
...
 
-class AdSensePerformanceWidget extends Component {
-	constructor( props ) {
-		super( props );
+const { useSelect } = Data;
 
-		this.state = {
-			twentyEightDays: false,
-			prev28Days: false,
+function AdSensePerformanceWidget( ) {
```
Comment (felixarntz) [inline assets/js/modules/adsense/components/dashboard/AdSensePerformanceWidget.js:46]: Regarding my above comment, the problem _may_ only be here: This component used to rely on `withData`, which is how the `handleDataSuccess` and `handleDataError` (which are not a great approach but are what is used right now) worked before. Now that that is gone, it causes the problem I mentioned above, where the `AdSenseDashboardZeroData` component is no longer showing. Let's either reverse the refactoring or can you suggest an alternative implementation based on what you have here that works?

**Review by felixarntz — CHANGES_REQUESTED** [review]
Overall this is looking good visually, but there are a few things missing, and I think the implementation is overly complex and partially different from the IB.

**Comment by ryanwelcher** [comment]
Closing this PR in favor of #2135

**Pull request closed without merging**